### PR TITLE
feat: add datadog test visibility harness to test ci

### PR DESCRIPTION
**Pull request by karrui**
Added `DD_API_KEY` secret to github

## Solution

This PR integrates Datadog Test Visibility into our CI workflow for both unit and end-to-end tests.

**Breaking Changes**

- [x] No - this PR is backwards compatible

**Improvements**:

- Added Datadog Test Visibility configuration for unit tests in the `test-studio` job
- Added Datadog Test Visibility configuration for end-to-end tests in the `end-to-end-tests` job
- Configured necessary environment variables to enable Datadog tracing for both Vitest and Playwright tests

## Tests

The existing unit and end-to-end tests will now be traced and reported to Datadog Test Visibility.

### CODE REVIEW

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/679?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#679** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/679?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#678** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/678?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#673** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/673?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,8 +69,23 @@ jobs:
         uses: xom9ikk/dotenv@v2
         with:
           mode: test
+      - name: Install dependencies
+        # Not using test:unit dependency in Turbo because Datadog Test Visibility requires only the test suite to be run
```
Comment (harishv7) [inline .github/workflows/ci.yml:73]: ps, i didn't understand this comment, can explain?

**Review by harishv7 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (karrui) [reply .github/workflows/ci.yml]: ya nvm it still doesn't work, man-in-the-middle does not like prisma generate haha. need do outside. dd-agent can only trace the test itself, not the full turbo pipeline

**Comment by karrui** [comment]
### Merge activity

* **Sep 23, 11:38 AM EDT**: @karrui started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/isomer/679).

**Pull request merged**